### PR TITLE
fix: fixed the registrations logic and modal messages

### PR DESCRIPTION
**Pull request by JoshM0101**
# Pull Request Review Comments

## Overview
- **Purpose of the PR**:
    - [ ] Fixed all registration logic so that it won't save any data if there is an conflict 
    - [ ] Modified the modal messages, made it a dynamic

### CODE REVIEW

File: client/src/pages/Expenses/ExpensesRegistration.tsx [hunk 1]
```diff
@@ -206,8 +206,30 @@ const ExpensesRegistration = () => {
       ])
     } catch (error) {
       if (error.response && error.response.status === 409) {
-        // Conflict: open the overwrite confirmation modal
-        setModalMessage(translate('alertMessageAbove', language));
+        const existingEntries = error.response.data.existingEntries;
```
Comment (edward-stalley-work) [inline client/src/pages/Expenses/ExpensesRegistration.tsx:209]: Would it be possible to clear form after successful submission? Right now information stays there.

Screen after successful submission:

Screenshot 2024-10-24 at 12 14 05

File: client/src/translations/general_words_en.json [hunk 2]
```diff
@@ -102,13 +105,15 @@
...
   "employeeExpensesValidation2":"Duplicate projects found. Please ensure each project is unique.",
   "employeeExpensesValidation3": "Please fill out all ${fieldsMessage} fields.",
   "employeeExpensesRemoveProject": "Project successfully removed",
-  "projectOverwriteMessage": "The data is already registered. Do you want to overwrite it?",
+  "projectOverwriteMessage": "The selected fields with: '${existingEntries}' already has data registered. Do you want to overwrite it?",
+  "projectNewEntry": " And save the new selected fields with: '${newEntries}'?",
   "updateMessage": "Are you sure you would like to update this data?",
-  "listScreen": "List Screen"
+  "listScreen": "List Screen",
+  "employeeExpensesDuplicateData": "There is already an existing data for ${employeeName}. Please choose a different name."
```
Comment (edward-stalley-work) [inline client/src/translations/general_words_en.json:118]: So lets make this a little more clear for employee expense.
Something like the following:

The selected project already exists for that ${year, month} for employee ${employeeName}

**Review by edward-stalley-work — CHANGES_REQUESTED** [review]
Looks good Joshua -  thank you.
Please see comments.

File: client/src/pages/Expenses/ExpensesRegistration.tsx (hunk 1 above)
Comment (JoshM0101) [reply client/src/pages/Expenses/ExpensesRegistration.tsx:209]: Fixed

File: client/src/translations/general_words_en.json (hunk 2 above)
Comment (JoshM0101) [reply client/src/translations/general_words_en.json]: Fixed

**Review by edward-stalley-work — APPROVED** [review]

**Pull request merged**